stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -371,43 +379,48 @@ U GetValue(
 }
 
 // Optical-port-specific version.
-template <typename T, typename U>
-U GetOpticalChannelValue(
+template <typename T, typename U, typename V>
+U GetValue(
     uint64 node_id, uint32 port_id, YangParseTree* tree,
-    const T& (OpticalChannelInfo::*instance_getter)() const,
-    bool (OpticalChannelInfo::*has_instance_getter)() const,
-    U (T::*inner_message_get_field_func)() const) {
+    const T& (DataResponse::*data_response_get_inner_message_func)() const,
+    bool (DataResponse::*data_response_has_inner_message_func)() const,
+    DataRequest::Request::Port* (
+        DataRequest::Request::*data_request_get_mutable_inner_message_func)(),
+    bool (T::*optical_channel_info_has_inner_message_func)() const,
```

Comment:
`data_req_has_inner_message_func` ?

Reply:
It's not for the request, but for the response inner message.

Reply:
`inner_msg_has_inner_msg_func`? (or maybe someting else?